test/lib/ansible_test/_util/controller/sanity/validate-modules/validate_modules/main.py

```diff
@@ -687,30 +687,57 @@ def _validate_ps_replacers(self):
         # get module list for each
         # check "shape" of each module name
 
-        module_requires = r'(?im)^#\s*requires\s+\-module(?:s?)\s*(Ansible\.ModuleUtils\..+)'
-        csharp_requires = r'(?im)^#\s*ansiblerequires\s+\-csharputil\s*(Ansible\..+)'
+        legacy_ps_requires = r'(?im)^#\s*Requires\s+\-Module(?:s?)\s+(Ansible\.ModuleUtils\..+)'
+        ps_requires = r'''(?imx)
+            ^\#\s*AnsibleRequires\s+-PowerShell\s+
+            (
+                # Builtin PowerShell module
+                (Ansible\.ModuleUtils\.[\w\.]+)
```

Comment:
Why does this pattern include `ModuleUtils`, but the one for C# below does not?

Reply:
Builtin C# module utils have always been `Ansible.Foo` and not `Ansible.ModuleUtils.Foo`. I'm unsure why we went that way but potentially it was to avoid conflicts with the older PowerShell ones.